### PR TITLE
Add new cases for multiple File Descriptor support

### PR DESCRIPTION
**Pull request by Yingshun**
This PR covers below cases:
live migration with parallel network connections:
    a. default network connections
    b. specified connection number
    c. boundary values of connection number
    d. without enabling parallel capability

Signed-off-by: Yingshun Cui <yicui@redhat.com>

### CODE REVIEW

**Comment by Yingshun** [comment]
**Test result:**(There are 2 types of error messages in current run, they are failed by BZ 1744530 and 1726937, will get them in blacklist of ci once this PR is approved.)
```
(.libvirt-ci-venv-ci-runtest-qInWGP) [root@yicui-host1 ~]#  avocado run --vt-type libvirt --vt-machine-type q35 multifd --vt-connect-uri qemu:///systemJOB ID     : e64d284652924197de209de5725e50fcfd554721
JOB LOG    : /root/avocado/job-results/job-2019-09-29T03.42-e64d284/job.log
 (01/27) type_specific.io-github-autotest-libvirt.virsh.migrate_options_shared.positive_test.multifd.customized_connection.set_minus.with_postcopy: PASS (176.05 s)
 (02/27) type_specific.io-github-autotest-libvirt.virsh.migrate_options_shared.positive_test.multifd.customized_connection.set_minus.without_postcopy: PASS (179.11 s)
 (03/27) type_specific.io-github-autotest-libvirt.virsh.migrate_options_shared.positive_test.multifd.customized_connection.set_min.with_postcopy: |PASS (184.62 s)
 (04/27) type_specific.io-github-autotest-libvirt.virsh.migrate_options_shared.positive_test.multifd.customized_connection.set_min.without_postcopy: |PASS (179.44 s)
 (05/27) type_specific.io-github-autotest-libvirt.virsh.migrate_options_shared.positive_test.multifd.customized_connection.set_normal.with_postcopy: /ERROR: No migration result is returned. (1137.57 s)
 (06/27) type_specific.io-github-autotest-libvirt.virsh.migrate_options_shared.positive_test.multifd.customized_connection.set_normal.without_postcopy: PASS (269.45 s)
 (07/27) type_specific.io-github-autotest-libvirt.virsh.migrate_options_shared.positive_test.multifd.customized_connection.set_max.with_postcopy: PASS (189.21 s)
 (08/27) type_specific.io-github-autotest-libvirt.virsh.migrate_options_shared.positive_test.multifd.customized_connection.set_max.without_postcopy: PASS (194.89 s)
 (09/27) type_specific.io-github-autotest-libvirt.virsh.migrate_options_shared.positive_test.multifd.defualt_connection.with_postcopy: ERROR: No migration result is returned. (1118.45 s)
 (10/27) type_specific.io-github-autotest-libvirt.virsh.migrate_options_shared.positive_test.multifd.defualt_connection.without_postcopy: PASS (251.32 s)
 (11/27) type_specific.io-github-autotest-libvirt.virsh.migrate_options_shared.positive_test.p2p_migration.multifd.customized_connection.set_minus.with_postcopy: PASS (189.49 s)
 (12/27) type_specific.io-github-autotest-libvirt.virsh.migrate_options_shared.positive_test.p2p_migration.multifd.customized_connection.set_minus.without_postcopy: PASS (181.81 s)
 (13/27) type_specific.io-github-autotest-libvirt.virsh.migrate_options_shared.positive_test.p2p_migration.multifd.customized_connection.set_min.with_postcopy: PASS (194.18 s)
 (14/27) type_specific.io-github-autotest-libvirt.virsh.migrate_options_shared.positive_test.p2p_migration.multifd.customized_connection.set_min.without_postcopy: PASS (181.09 s)
 (15/27) type_specific.io-github-autotest-libvirt.virsh.migrate_options_shared.positive_test.p2p_migration.multifd.customized_connection.set_normal.with_postcopy: ERROR: No migration result is returned. (1115.57 s)
 (16/27) type_specific.io-github-autotest-libvirt.virsh.migrate_options_shared.positive_test.p2p_migration.multifd.customized_connection.set_normal.without_postcopy: PASS (270.54 s)
 (17/27) type_specific.io-github-autotest-libvirt.virsh.migrate_options_shared.positive_test.p2p_migration.multifd.customized_connection.set_max.with_postcopy: PASS (192.73 s)
 (18/27) type_specific.io-github-autotest-libvirt.virsh.migrate_options_shared.positive_test.p2p_migration.multifd.customized_connection.set_max.without_postcopy: PASS (193.34 s)
 (19/27) type_specific.io-github-autotest-libvirt.virsh.migrate_options_shared.positive_test.p2p_migration.multifd.defualt_connection.with_postcopy: ERROR: No migration result is returned. (1127.06 s)
 (20/27) type_specific.io-github-autotest-libvirt.virsh.migrate_options_shared.positive_test.p2p_migration.multifd.defualt_connection.without_postcopy: PASS (260.04 s)
 (21/27) type_specific.io-github-autotest-libvirt.virsh.migrate_options_shared.negative_test.multifd.without_parallel.without_postcopy: PASS (168.00 s)
 (22/27) type_specific.io-github-autotest-libvirt.virsh.migrate_options_shared.negative_test.multifd.inavalid_connect_num.set_zero.without_postcopy: PASS (172.10 s)
 (23/27) type_specific.io-github-autotest-libvirt.virsh.migrate_options_shared.negative_test.multifd.inavalid_connect_num.set_cross_border.without_postcopy: PASS (172.06 s)
 (24/27) type_specific.io-github-autotest-libvirt.virsh.migrate_options_shared.negative_test.p2p_migration.multifd.without_parallel.without_postcopy: PASS (167.00 s)
 (25/27) type_specific.io-github-autotest-libvirt.virsh.migrate_options_shared.negative_test.p2p_migration.multifd.inavalid_connect_num.set_zero.without_postcopy: PASS (172.42 s)
 (26/27) type_specific.io-github-autotest-libvirt.virsh.migrate_options_shared.negative_test.p2p_migration.multifd.inavalid_connect_num.set_cross_border.without_postcopy: PASS (170.04 s)
 (27/27) type_specific.io-github-autotest-libvirt.virsh.migrate_options_shared.negative_test.tunnelled_migration.multifd.without_postcopy: FAIL: Can not find the expected patterns 'unsupport' in output 'error: operation failed: domain is not running' (155.38 s)
RESULTS    : PASS 22 | ERROR 4 | FAIL 1 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0
JOB TIME   : 8967.68 s

```

File: libvirt/tests/src/migration/migrate_options_shared.py [hunk 1]
```diff
@@ -625,6 +625,24 @@ def drop_network_connetion(block_time):
...
+
+        :param exp_num: expected number of the ESTABLISHED connections
+        :param port: port to be checked
+        :raise: test.fail if the connection number is not equal to $exp_num
+        """
+        if postcopy_options:
+            set_migratepostcopy()
+
+        cmd = "netstat -tunap|grep %s |grep ESTABLISHED | wc -l" % port
+        result = remote.run_remote_cmd(cmd, cmd_parms, runner_on_target)
```
Comment (dzhengfy) [inline libvirt/tests/src/migration/migrate_options_shared.py]: Check result exit status

File: libvirt/tests/src/migration/migrate_options_shared.py [hunk 2]
```diff
@@ -1072,6 +1093,24 @@ def check_migration_res(result):
...
             extra = "%s %s" % (extra, cache)
 
+        parallel_cn_nums = params.get("parallel_cn_nums")
+        if parallel_cn_nums or extra.count("parallel"):
+            if not libvirt_version.version_compare(5, 2, 0):
+                test.cancel("This libvirt version doesn't support "
+                            "multifd feature.")
+
+            if parallel_cn_nums is not None:
+                parallel_connections = params.get("parallel_connections",
```
Comment (dzhengfy) [inline libvirt/tests/src/migration/migrate_options_shared.py]: 'parallel_connections' can be directly configured in virsh_migrate_extra for simplicity.

Comment (Yingshun) [reply libvirt/tests/src/migration/migrate_options_shared.py]: updated.

File: libvirt/tests/src/migration/migrate_options_shared.py (hunk 1 above)
Comment (Yingshun) [reply libvirt/tests/src/migration/migrate_options_shared.py]: @dzhengfy updated with ignore_status=False

File: libvirt/tests/src/migration/migrate_options_shared.py [hunk 3]
```diff
@@ -1072,6 +1094,21 @@ def check_migration_res(result):
             cache = get_usable_compress_cache(memory.get_page_size())
             extra = "%s %s" % (extra, cache)
 
+        parallel_cn_nums = params.get("parallel_cn_nums")
+        if parallel_cn_nums or extra.count("parallel"):
+            if not libvirt_version.version_compare(5, 2, 0):
+                test.cancel("This libvirt version doesn't support "
```
Comment (dzhengfy) [inline libvirt/tests/src/migration/migrate_options_shared.py]: Canceling a case is better to be executed asap which is to save much time to start vm and do some unnecessary configuration. So I suggest we can move this version test closed to and under 'try:' clock. What do you think of?

Comment (Yingshun) [reply libvirt/tests/src/migration/migrate_options_shared.py]: @dzhengfy  Updated.

**Comment by Yingshun** [comment]
@chloerh please help review.
